core/src/main/java/com/digitalpebble/stormcrawler/parse/filter/MD5SignatureParseFilter.java

```diff
@@ -33,18 +33,39 @@
 /**
  * Computes a signature for a page, based on the binary content or text. If the
  * content is empty, the URL is used.
+ *
+ * Configuration properties:
+ * <dl>
+ * <dt>useText</dt>
+ * <dd>compute signature on plain text, instead of binary content</dd>
+ * <dt>keyName</dt>
+ * <dd>name of the metadata field to hold the signature (default:
+ * &quot;signature&quot;)</dd>
+ * <dt>keyNameCopy</dt>
+ * <dd>name of the metadata field to hold a temporary copy of the signature used
+ * to decide by signature comparison whether the document has changed. If not
+ * defined, the signature is not copied.</dd>
+ * </dl>
+ *
  */
 public class MD5SignatureParseFilter extends ParseFilter {
 
     private String key_name = "signature";
 
     private boolean useText = false;
 
+    private String copyKeyName = null;
+
     @Override
     public void filter(String URL, byte[] content, DocumentFragment doc,
             ParseResult parse) {
         ParseData parseData = parse.get(URL);
         Metadata metadata = parseData.getMetadata();
+        if (copyKeyName != null) {
```

Comment:
StringUtils.isNotBlank would be better\: the value could be an empty String or spaces, we wouldn't want that as key and it probably isn't what the user intended to set